mgopersist/persist.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// Package mgopersist defines cached MongoDB-backed global persistent storage for
+// key-value pairs.
+package mgopersist
+
+import (
+	"sync"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var ErrNotFound = errgo.New("persistent data entry not found")
+
+type entry struct {
```

Comment:
I started changing it to use that package, but I'm not convinced it's entirely
appropriate here. Most of the complexity of utils/cache comes from dealing
with a potentially unbounded set of keys, which is something I don't think
we need to deal with here. I think I prefer it being simpler and free of that
dependency for the time being.